src/app/users/users.component.ts

```diff
@@ -125,7 +125,10 @@ export class UsersComponent implements OnInit, AfterViewInit {
 
   addRole(user) {
     // If user has no previous role, add learner role
-    const selectedRolesArray = this.selectedRolesMap.get(user.name) || [ 'learner' ];
+    let selectedRolesArray = this.selectedRolesMap.get(user.name);
+    if ( this.selectedRolesMap.get(user.name) === undefined || this.selectedRolesMap.get(user.name).length === 0 ) {

```

Comment:
can be simplify it using if (!!this.selectedRolesMap.get(user.name)) . Could you please check it once ?